CONTRIBUTING.md

```diff
@@ -30,6 +30,16 @@ After that, ensure all lints, tests, and builds are successful by running:
 - `cargo test --all-features --workspace`
 - `cargo build --all-features --all-targets --workspace`
 
+## Crate names
+Cuprate's crates (libraries) follows these naming patterns/rules:
+
+| Pattern                                               | Name             | Example |
+|-------------------------------------------------------|------------------|---------|
+| Crates defining Monero related behavior               | `monero` prefix  | `monero-consensus`
+| Crates specifically for Cuprate's implementation      | `cuprate` prefix | `cuprate-blockchain`
+| Monero related code re-written for Cuprate purposes   | `cuprate` suffix | `levin-cuprate`, `cryptonight-cuprate`
```

Comment:
Is there a better description for the difference between the 1st and 3rd rule?

I.e. what does the `-cuprate` suffix mean?

Reply:
1st: Crates defining Monero intrinsic behavior that is software-agnostic (can be used outside of Cuprate).
3rd: General purpose crates re-implementing monero-related code for Cuprate specifically (usually by redefining API).

Reply:
Also:
```suggestion
| Crates defining Monero related behavior               | `monero` prefix  | `monero-consensus`
| Crates specific to Cuprate's implementation      | `cuprate` prefix | `cuprate-blockchain`
| Monero related code re-written for Cuprate purposes   | `cuprate` suffix | `levin-cuprate`, `cryptonight-cuprate`
```

Reply:
> I.e. what does the -cuprate suffix mean?

I have used it when a better alternative isn't available, i.e. `-cuprate` pretty much means the same as no prefix.

btw I changed monero-consensus to cuprate-consensus-rules in #65 could probably change it to monero-consensus-rules so it follows this spec though

Reply:
> 1st: Crates defining Monero intrinsic behavior that is software-agnostic (can be used outside of Cuprate).

Is this true currently? Can all `monero-` crates be used without pulling `cuprate-` dependencies?

> I have used it when a better alternative isn't available, i.e. -cuprate pretty much means the same as no prefix.

Open to suggestions, we can change the spec to anything (I just want it formalized).

Reply:
A simpler option is to remove prefix/suffixes from directory names and always prefix `cuprate-` to all crate names, like [`reth`](https://github.com/paradigmxyz/reth/tree/main/crates).

E.g.
- `database/` -> `cuprate-database`
-  `levin/` -> `cuprate-levin`
- `epee-encoding/` -> `cuprate-epee-encoding`
- `monero-wire/` -> `cuprate-wire` or `cuprate-monero-wire`

This is clearer if/when uploading to crates.io as well. I think I'd prefer this way actually.

Reply:
This would be nice and simple, however we have:

`monero-p2p` a lower level p2p crate

`cuprate-p2p` a higher level crate built off `monero-p2p`

I don't really know what we would change `monero-p2p` to.

Reply:
`cuprate-monero-p2p` or `cuprate-p2p-core`? Examples of `-core` suffix representing lower-level crates:
- https://docs.rs/rayon_core
- https://docs.rs/rand_core
- https://docs.rs/parking_lot_core
- https://docs.rs/tracing_core
- https://docs.rs/futures_core
- https://docs.rs/axum_core

Reply:
yeah `cuprate-p2p-core` is a good idea